drivers/rn2xx3/rn2xx3.c

```diff
@@ -0,0 +1,331 @@
+        dev->resp_buf[dev->resp_size++] = '\0';
+        if (dev->int_state == RN2XX3_INT_STATE_MAC_RX_MESSAGE) {
+            /* RX state: closing RX buffer */
+            dev->rx_buf[(dev->rx_size + 1) / 2] = 0;
+            if (netdev->event_callback) {
+                netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+            }
+        }
+        else if (dev->int_state == RN2XX3_INT_STATE_MAC_TX) {
+            /* still in TX state: transmission complete but no data received */
+            if (netdev->event_callback) {
+                netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+            }
+        }
+        dev->resp_size = 0;
+        dev->rx_size = 0;
+        dev->resp_done = 1;
+        mutex_unlock(&(dev->resp_lock));
+    }
+    else {
+        switch (dev->int_state) {
+            /* A successful mac TX command expects 2 kinds of replies:
+                 * - mac_tx_ok: transmission done, no data received
+                 * - mac_rx <port> <data>: transmission done, some data
+                 *                         received on port <port>. */
+            case RN2XX3_INT_STATE_MAC_TX:
+                /* some data are available */
+                dev->resp_buf[dev->resp_size++] = c;
+                /* if module response ends with 'rx ' after 8 received chars,
+                   the module starts receiving data */
+                if (dev->resp_size == 8 && dev->resp_buf[4] == 'r'
+                    && dev->resp_buf[5] == 'x' && dev->resp_buf[6] == ' ') {
+                    /* next received chars correspond to the port number */
+                    dev->int_state = RN2XX3_INT_STATE_MAC_RX_PORT;
+                }
+                break;
+
+            case RN2XX3_INT_STATE_MAC_RX_PORT:
+                dev->resp_buf[dev->resp_size++] = c;
+                if (c == ' ') {
+                    dev->int_state = RN2XX3_INT_STATE_MAC_RX_MESSAGE;
+                }
+                break;
+
+            case RN2XX3_INT_STATE_MAC_RX_MESSAGE:
+                /* read and convert RX data (received in hex chars) */
+                if (c == ' ') {
+                    dev->resp_buf[dev->resp_size++] = c;
+                    break;
+                }
+                dev->rx_tmp_buf[dev->rx_size % 2] = c;
+
+                /* We convert pairs of hex character to bytes on the fly to
+                   save space in memory */
+                if (dev->rx_size != 0 && dev->rx_size % 2) {
+                    rn2xx3_hex_to_bytes(dev->rx_tmp_buf,
+                                        &dev->rx_buf[(dev->rx_size - 1) / 2]);
+                }
+                dev->rx_size++;
+                break;
+
+            default:
+                dev->resp_buf[dev->resp_size++] = c;
+                break;
+        }
+    }
+}
+
+static void _sleep_timer_cb(void *arg)
+{
+    rn2xx3_t *dev = (rn2xx3_t *)arg;
+    dev->int_state = RN2XX3_INT_STATE_IDLE;
+    DEBUG("[rn2xx3] exit sleep\n");
```

Comment:
When I activate `#define ENABLE_DEBUG    (1)` in the context of an ISR the stack is too small. Maybe increasing the ISR stack would solve the issue.